tools/Azure.Mcp.Tools.Monitor/src/Commands/ActivityLog/ActivityLogListCommand.cs

```diff
@@ -0,0 +1,153 @@
+          Optional options:
+        - --resource-type: The resource type (e.g., 'Microsoft.Storage/storageAccounts') for disambiguation
+        - --hours: Number of hours to look back
+        - --event-level: Filter by event level (Critical, Error, Informational, Verbose, Warning)
+        - --top: Maximum number of logs to return
+        """;
+
+    public override string Title => CommandTitle;
+
+    public override ToolMetadata Metadata => new()
+    {
+        Destructive = false,
+        OpenWorld = true,
+        Idempotent = true,
+        ReadOnly = true,
+        Secret = false,
+        LocalRequired = false
+    };
+
+    protected override void RegisterOptions(Command command)
+    {
+        base.RegisterOptions(command);
+        command.Options.Add(OptionDefinitions.Common.ResourceGroup.AsOptional());
+        command.Options.Add(_resourceNameOption);
+        command.Options.Add(_resourceTypeOption);
+        command.Options.Add(_hoursOption);
+        command.Options.Add(_eventLevelOption);
+        command.Options.Add(_topOption);
+    }
+
+    protected override ActivityLogListOptions BindOptions(ParseResult parseResult)
+    {
+        var options = base.BindOptions(parseResult);
+        options.ResourceName = parseResult.GetValueOrDefault(_resourceNameOption);
+        options.ResourceType = parseResult.GetValueOrDefault(_resourceTypeOption);
+        options.Hours = parseResult.GetValueOrDefault(_hoursOption);
+        options.EventLevel = parseResult.GetValueOrDefault(_eventLevelOption);
+        options.Top = parseResult.GetValueOrDefault(_topOption);
+        return options;
+    }
+
+    public override async Task<CommandResponse> ExecuteAsync(CommandContext context, ParseResult parseResult)
+    {
+        // Required validation step
+        if (!Validate(parseResult.CommandResult, context.Response).IsValid)
+        {
+            return context.Response;
+        }
+
+        var options = BindOptions(parseResult);
+
+        try
+        {
+            // Get the Monitor service from DI
+            var service = context.GetService<IMonitorService>();
+
+            // Call service operation with required parameters
+            var results = await service.ListActivityLogs(
+                options.Subscription!,
+                options.ResourceName!,
+                options.ResourceGroup,
+                options.ResourceType,
+                options.Hours ?? 24.0,
+                options.EventLevel,
+                options.Top ?? 10,
+                options.Tenant,
+                options.RetryPolicy);
+
+            // Set results if any were returned
+            context.Response.Results = results?.Count > 0 ?
+                ResponseResult.Create(
+                    new ActivityLogListCommandResult(results),
+                    MonitorJsonContext.Default.ActivityLogListCommandResult) :
+                null;
```

Comment:
Return a result with an explicit empty array instead of null.
> // For enumerable returns, coalesce null into an empty enumerable.
in https://github.com/microsoft/mcp/blob/main/servers/Azure.Mcp.Server/docs/new-command.md